Adjust ImageMeta dimensions based on Exif orientation data

Insprill opened the pull request:
The current handling of image dimensions when rotated with Exif data is relatively poor, as it's only handled in `ImageOp.perform`. This means that other image operations such as `get_image_metadata` and `resize_image` will be operating with the expected width and height flipped.

This proved to be problematic for a macro I had written to automatically create multiple scaled versions of images. It works fine for landscape images, but squishes portrait images into the size as if it wasn't rotated at all.
<details>
<summary>Macro code</summary>

```html
{% macro responsive_img(path, alt="", sizes="100vw", priority="auto", class="") -%}
  {% set img_fmt = "avif" %}
  {% set img_qlt = 70 %}

  {% set meta = get_image_metadata(path=path) %}
  {% set w1 = meta.width %}
  {% set w2 = meta.width / 2 | int %}
  {% set w3 = meta.width / 4 | int %}
  {% set w4 = meta.width / 8 | int %}
  {% set w5 = meta.width / 16 | int %}
  {% set w6 = meta.width / 32 | int %}

  {% set i1 = resize_image(path=path, width=w1, op="fit_width", format=img_fmt, quality=img_qlt) %}
  {% set i2 = resize_image(path=path, width=w2, op="fit_width", format=img_fmt, quality=img_qlt) %}
  {% set i3 = resize_image(path=path, width=w3, op="fit_width", format=img_fmt, quality=img_qlt) %}
  {% set i4 = resize_image(path=path, width=w4, op="fit_width", format=img_fmt, quality=img_qlt) %}
  {% set i5 = resize_image(path=path, width=w5, op="fit_width", format=img_fmt, quality=img_qlt) %}
  {% set i6 = resize_image(path=path, width=w6, op="fit_width", format=img_fmt, quality=img_qlt) %}

  <img
    class = "{{ class }}"
    src="{{ i6.url }}"
    srcset="
      {{ i1.url }} {{ w1 }}w,
      {{ i2.url }} {{ w2 }}w,
      {{ i3.url }} {{ w3 }}w,
      {{ i4.url }} {{ w4 }}w,
      {{ i5.url }} {{ w5 }}w,
      {{ i6.url }} {{ w6 }}w,
    "
    sizes="{{ sizes }}"
    alt="{{ alt }}"
    loading="lazy"
    fetchpriority="{{ priority }}"
  />
{%- endmacro %}
```

</details>

With the above macro and Zola v0.20.0, the following image is output:
![DSC00819 14678d46ab9ae7d2](https://github.com/user-attachments/assets/d765b64b-950b-48a5-b29e-5a4a195a02b4)

With this PR, the output is as expected:
![DSC00819 854761b8da61362d](https://github.com/user-attachments/assets/7958dff3-70da-468d-92e6-b59935bc2c5d)

I initially wrote this to swap only the dimensions in the `read_image_metadata` default match arm. This fixed the issue for `get_image_metadata` but did not fix the issue for `resize_image`, which would still result in the same "before" image regardless of whether I gave `get_image_metadata`'s width or height to the width parameter. This was because it parsed its own copy of `ImageMeta` without correcting the rotation values. Because of this, and to avoid placing rotation corrections all over the codebase, I only applied it in `ImageMeta.read`, which covers both of my use cases, and potentially others that may otherwise have been missed. This doesn't affect the rotation handling in `ImageOp.perform` since it reads the image file directly, without the use of `ImageMeta`.

I haven't updated the documentation since I didn't see any mentions of rotation handling there. With this PR, it should be as expected, so I don't believe a mention is necessary, although I'll gladly write something up if requested.

This is my first time contributing to Zola, so please let me know if I should've done anything differently! I also haven't been working with Zola for too long, so I can't be certain this doesn't have any adverse effects, but everything appeared to be functioning fine on my site, and all tests are passing, so it would seem everything is good. A thorough review and testing should be done prior to merging, though, just to be safe.

Closes #2818

Sanity check:

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes
(Delete or ignore this section for documentation changes)

* [x] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [ ] Have you created/updated the relevant documentation page(s)?

Insprill committed:
Adjust ImageMeta dimensions based on Exif data

Keats commented on components/imageproc/src/meta.rs:
```diff
@@ -21,7 +23,11 @@ impl ImageMeta {
         let reader = ImageReader::open(path).and_then(ImageReader::with_guessed_format)?;
         let format = reader.format();
-        let size = reader.into_dimensions()?;
+        let mut size = reader.into_dimensions()?;
```
can we somehow get the exif metadata from the image object? So we don't have to read the file twice

Insprill replied:
Yup, image-rs has a function on `ImageDecoder` to get the raw Exif data. I made `get_orientation` take the raw Exif data and updated all the upstream usages (ImageMeta, ImageOp, tests) to get it this way.

Insprill committed:
Get image Exif data from the decoder

This lets us avoid reading the file from disk twice - once to get the
dimensions/contents, and once to get the image metadata. Now we create
one decoder, get all the data we need from it, and turn it into a
DynamicImage directly.

Insprill committed:
Get image size directly from ImageDecoder in get_rotated_size_test
